emacs-the-best-python-editor/init.el

```diff
@@ -0,0 +1,72 @@
+;; .emacs.d/init.el
+
+;; ===================================
+;; MELPA Package Support
+;; ===================================
+;; Enables packaging support
+(require 'package)
+
+;; Adds the Melpa archive to the list of available repositories
+(add-to-list 'package-archives
+             '("melpa" . "http://melpa.org/packages/") t)
+
+;; Initializes the package infrastructure
+(package-initialize)
+
+;; If there are no archived package contents, refresh them
+(when (not package-archive-contents)
+  (package-refresh-contents))
+
+;; Installs packages
+;;
+;; myPackages contains a list of package names
+(defvar myPackages
+  '(better-defaults                 ;; Setup some better Emacs defaults
+    elpy                            ;; Emacs Lisp Python Environment
+    ein                             ;; Emacs iPython Notebook
+    flycheck                        ;; On the fly syntax checking
+    py-autopep8                     ;; Run autopep8 on save
+    blacken                         ;; Black formatting on save
+    magit                           ;; Git integration
+    material-theme                  ;; Theme
+    )
+  )
+
+;; Scans the list in myPackages
+;; If the package listed is not already installed, install it
+(mapc #'(lambda (package)
+          (unless (package-installed-p package)
+            (package-install package)))
+      myPackages)
+
+;; ====================================
+;; Basic Customization
+;; ====================================
+
+(setq inhibit-startup-message t)  ;; Hide the startup message
+(load-theme 'material t)          ;; Load material theme
+(global-linum-mode t)             ;; Enable line numbers globally
+
+;; ====================================
+;; DEVELOPMENT SETUP
+;; ====================================
+;; Enable elpy
+(elpy-enable)
+
+;; Use IPython for REPL
+;; (setq python-shell-interpreter "jupyter"
```

Comment:
The IPython stuff is commented out because not everyone uses it. Enabling IPython replaces the REPL completely, so I opted to comment it out. If someone wants/needs it, they can uncomment it.

As for autopep8, same argument - you can enable one autoformatter, and I opted for black. Leaving the option there allows someone to change it.

Reply:
Ah, okay. Makes sense :+1: